WordPress/Jetpack/Classes/System/JetpackWindowManager.swift

```diff
@@ -2,64 +2,74 @@ import Combine
 import Foundation
 
 class JetpackWindowManager: WindowManager {
-    /// receives migration flow updates in order to dismiss it when needed.
+
+    /// Receives migration flow updates in order to dismiss it when needed.
     private var cancellable: AnyCancellable?
 
+    /// Reference to the presented migration view controller.
+    ///
+    /// Used to prevent a new migration view controller instance to be presented if
+    /// an existing one is already presented.
+    private weak var migrationViewController: UIViewController?
+
     /// Migration events tracking
     private let migrationTracker = MigrationAnalyticsTracker()
 
-    var shouldImportMigrationData: Bool {
-        return FeatureFlag.contentMigration.enabled
-        && !AccountHelper.isLoggedIn
-        && !UserPersistentStoreFactory.instance().isJPContentImportComplete
-    }
-
     override func showUI(for blog: Blog?, animated: Bool = true) {
-        if AccountHelper.isLoggedIn {
-            showAppUI(for: blog)
+        // Show migration flow if eligible
+        let migrationStarted = startMigrationFlowIfNeeded(blog)
+        if migrationStarted {
             return
         }
 
-        guard FeatureFlag.contentMigration.enabled else {
-            showSignInUI()
+        // Show App UI if user is logged in
+        if AccountHelper.isLoggedIn {
+            self.showAppUI(for: blog)
             return
         }
 
-        self.migrationTracker.trackContentImportEligibility(eligible: shouldImportMigrationData)
-        shouldImportMigrationData ? importAndShowMigrationContent(blog) : showSignInUI()
+        // Show Sign In UI if previous conditions are false
+        self.showSignInUI()
     }
+}
 
-    func importAndShowMigrationContent(_ blog: Blog? = nil) {
-        self.migrationTracker.trackWordPressMigrationEligibility()
+// MARK: - Migration Related Properties and Methods
 
-        DataMigrator().importData() { [weak self] result in
-            guard let self else {
-                return
-            }
+extension JetpackWindowManager {
 
-            switch result {
-            case .success:
-                self.migrationTracker.trackContentImportSucceeded()
-                UserPersistentStoreFactory.instance().isJPContentImportComplete = true
-                NotificationCenter.default.post(name: .WPAccountDefaultWordPressComAccountChanged, object: self)
-                self.showMigrationUIIfNeeded(blog)
-                self.sendMigrationEmail()
-            case .failure(let error):
-                self.migrationTracker.trackContentImportFailed(reason: error.localizedDescription)
-                self.handleMigrationFailure(error)
-            }
+    @discardableResult
+    func startMigrationFlowIfNeeded(_ blog: Blog? = nil) -> Bool {
+        guard shouldStartMigrationFlow else {
+            return false
         }
+        self.importAndShowMigrationContent(blog)
+        return true
     }
```

Comment:
This is just a convenient method to avoid calling this code in [JetpackWindowManager.showUI](https://github.com/wordpress-mobile/WordPress-iOS/pull/19888/files#diff-bbbf5bdbd178be9be3ed7ba8cea3ae4a19b0b1f28cacdd647d2704014f96b98bR20) and [AppDelegate](https://github.com/wordpress-mobile/WordPress-iOS/pull/19888/files#diff-a0d929b4f010704e861fa3bca33fd60c2accdbeed6669393b613d9390a7b2ae5R201):

```swift
if shouldStartMigrationFlow {
    importAndShowMigrationContent(blog)
}
```